### PR TITLE
feat: #5105 add pathname at the end of dropdown language url

### PR DESCRIPTION
**Pull request by cs09g**
Added pathname to keep the consistency of reading while changing language.

Closes https://github.com/webpack/webpack.js.org/issues/5105

### CODE REVIEW

**Comment by chenxsan** [comment]
Really nice feature. However there is one problem, say I'm browsing `/concepts/plugins/#usage` and want to check the respective Chinese version, clicking the button would bring me to `chinese-site/concepts/plugins/` instead of `chinese-site/concepts/plugins/#usage`.

I think we should retain the URI fragment as well, just like how we did when changing the version of webpack documentation:

CleanShot 2021-06-24 at 12 11 58@2x

Here's the code https://github.com/webpack/webpack.js.org/blob/master/src/components/Sidebar/Sidebar.jsx#L25 just in case you want to take a look.

**Comment by chenxsan** [comment]
Btw, can you sign cla here https://cla.js.foundation/webpack/webpack.js.org?pullRequest=5108?

**Comment by cs09g** [comment]
@chenxsan I've signed! 😀

**Comment by cs09g** [comment]
> Really nice feature. However there is one problem, say I'm browsing `/concepts/plugins/#usage` and want to check the respective Chinese version, clicking the language button would bring me to `chinese-site/concepts/plugins/` instead of `chinese-site/concepts/plugins/#usage`.
> 
> I think we should retain the URI fragment as well, just like how we did when changing the version of webpack documentation:
> 
> CleanShot 2021-06-24 at 12 11 58@2x
> 
> Here's the code https://github.com/webpack/webpack.js.org/blob/master/src/components/Sidebar/Sidebar.jsx#L25 just in case you want to take a look.

Good catch! what if I pass `location.hash` as a prop to use it on `Navigation`?

**Comment by chenxsan** [comment]
I just found a bug in the current implementation.

Here's how to reproduce:

Open https://webpack-js-org-git-fork-cs09g-feature-issue-bdc88a-webpack-docs.vercel.app/concepts/entry-points/#multi-page-application, then change the version. You'll see the URI fragment is missing.

**Comment by cs09g** [comment]
@chenxsan Hmm, the hash is missing. Is that based on the last commit on this PR? Or is something different on the build with on localhost?

**Comment by chenxsan** [comment]
We have this preview https://github.com/webpack/webpack.js.org/pull/5108#issuecomment-867109431 synced with the latest commit of current pull request, so the preview should be up to date.

The build is actually a server-side rendered first, then hydrated on client-side app. So, yes, it's not the same as the dev one on localhost.

**Comment by cs09g** [comment]
@chenxsan I've pushed one more commit after the preview generated. I guess it's fine with the next deployment. check this out. https://webpack-js-org-1ozc7s86g-webpack-docs.vercel.app/

**Comment by chenxsan** [comment]
Neither does it work. You can open https://webpack-js-org-1ozc7s86g-webpack-docs.vercel.app/concepts/entry-points/#multi-page-application to inspect.

But I did notice the discrepancy here:

CleanShot 2021-06-25 at 09 58 38@2x

The `href` prop in React virtual DOM is up to date, however the real DOM is stale.

So the problem could be `hash` is client side only, and it's not hydrated on client side causing the issue. How about handling it with `useEffect`?

**Comment by cs09g** [comment]
I got it. I could check this out with `yarn build-test' on localhost. By the way, It's a class component. It needs to be refactored.

**Comment by chenxsan** [comment]
@cs09g Seems the CLA is not signed, can you double check?

File: src/components/Navigation/Navigation.jsx [hunk 1]
```diff
@@ -26,93 +26,27 @@ import StackOverflowIcon from '../../styles/icons/stack-overflow.svg';
...
+function Navigation({
+  pathname,
+  hash = '',
+  links,
+  toggleSidebar,
+  theme,
+  switchTheme,
+}) {
+  const themeSwitcher = () => switchTheme(theme === DARK ? LIGHT : DARK);
+  const [locationHash, setLocationHash] = useState('');
```
Comment (chenxsan) [inline src/components/Navigation/Navigation.jsx:47]:
```suggestion
  const [locationHash, setLocationHash] = useState(hash);
```
Can we initialize it with a default value from prop? Any possible flaw?

**Comment by cs09g** [comment]
@chenxsan I did. Some of my commits pushed as different emails with Github. Should I re-commit with the email that I signed in CLA?
image

File: src/components/Navigation/Navigation.jsx (hunk 1 above)
Comment (cs09g) [reply src/components/Navigation/Navigation.jsx:47]: I was afraid of `undefined` as a string. but actually, that's not going to happen. I'll fix it

**Comment by chenxsan** [comment]
> @chenxsan I did. Some of my commits pushed as different emails with Github. Should I re-commit with the email that I signed in CLA?
> image

Yeah, the bot is not that smart to handle it.

**Review by chenxsan — APPROVED** [review]
Really great enhancement! Now users can toggle between different languages seamlessly. Thanks for your contribution.

**Comment by cs09g** [comment]
@chenxsan Thanks for the warm review!

**Pull request merged**